server/neptune/workflows/internal/deploy/revision/queue/deployer.go

```diff
@@ -86,12 +87,39 @@ func (p *Deployer) Deploy(ctx workflow.Context, requestedDeployment terraformWor
 		logger.Error(ctx, "unable to persist deployment, proceeding with in-memory value", key.ErrKey, persistErr)
 	}
 
+	// rebase open PRs
+	// worker uses the returned error types to perform some follow up tasks, so instead of propagating the rebase error back, we configure
+	// infinite retries and log the error to ensure we rebase open PRs before deploying another change in this root
+	if rebaseErr := p.rebaseOpenPRsForRoot(ctx, requestedDeployment.Repo); rebaseErr != nil {
```

Comment:
Our current setup uses the returned error type to do some additional work in the caller. If the child workflow errors out due to a `TFClientError` and the rebase flow also errors out, we would want to return the child workflow error, not the rebase flow error. So, I have configured infinite retries for the rebase flow to ensure we rebase all open PRs before deploying any changes for this root, and logged the error. Lmk if you have better suggestions.

Reply:
Maybe it shouldn't belong within the deployer if it's not how you want to define a deployment's success/failure, thoughts on putting it into the worker? Or is there a specific reason you didn't want to propagate a rebase error type back?

Also, I don't think infinite retries without some form of eventual failure is ideal.

Reply:
>Maybe it shouldn't belong within the deployer if it's not how you want to define a deployment's success/failure, > thoughts on putting it into the worker?

A deployment success/failure should not depend on whether we successfully rebased all open PRs for the root since it's a follow up work, rather than a part of the deployment. We could put it in the worker but a worker shouldn't technically need to know about the follow up tasks either imo. 

> Or is there a specific reason you didn't want to propagate a rebase error type back?

The reason I'm not propagating the rebase error is the same reason why we don't propagate the `persistError` when persisting a deployment; we check the error type in the worker to decide the reason why the terraform workflow failed. If we propagate the rebase error, we will need to wrap the `WorkflowError`  but we'll need to add support in the worker to parse the wrapped error. 

Rebasing an open PR is necessary before deploying any more changes for a root, similar to how we need to persist the deployment after a deploy. 

> Also, I don't think infinite retries without some form of eventual failure is ideal.

I agree that we need to have some eventual failure with infinite retries. I have added a `MaximumInterval` to 8 hours which should be more than enough for our API ratelimits to revive if we run out of it.